<?xml version="1.0"?>

<robot name="scout_v2"
    xmlns:xacro="http://ros.org/wiki/xacro">

    <xacro:arg name="robot_namespace" default="/" />

    <xacro:include filename="$(find robot_description)/urdf/scout_v2.xacro" />

    <material name="black">
    <color rgba="0.0 0.0 0.0 0.7"/>
    </material>

    <material name="blue">
    <color rgba="0.0 0.0 0.8 0.8"/>
     </material>

    <!-- Add IMU -->
    <link name="imu_link">
        <visual>
            <origin xyz="0.0 0.0 0.0"/>
            <geometry>
                <box size="0.035 0.035 0.017"/>
            </geometry>
            <material name="black"/>
        </visual>
    </link>

    <joint name="imu_joint" type="fixed">
        <parent link="base_link"/>
        <child link="imu_link"/>
        <origin rpy="0 0 0" xyz="-0.01 0.0 0.10"/>
    </joint>

    <!-- Add GPS Receiver -->
    <link name="gps_link">
        <visual>
            <origin xyz="0.0 0.0 0.0"/>
            <geometry>
                <box size="0.05 0.045 0.024"/>
            </geometry>
            <material name="black"/>
        </visual>
    </link>

    <joint name="gps_joint" type="fixed">
        <parent link="base_link"/>
        <child link="gps_link"/>
        <origin rpy="0 0 0" xyz="-0.34 0.0 0.10"/>
    </joint>

    <!-- Add Lidar -->
    <link name="laser_link">
        <visual>
            <origin xyz="0.0 0.0 0.0"/>
            <geometry>
                <cylinder length="0.087" radius="0.08"/>
            </geometry>
            <material name="blue"/>
        </visual>
    </link>

    <joint name="base_laser_joint" type="fixed">
        <parent link="base_link"/>
        <child link="laser_link"/>
        <origin rpy="0 0 0" xyz="0.075 0.0 0.51"/>
    </joint>

</robot>
